gdal/ogr/ogrsf_frmts/sosi/ogrsosilayer.cpp

```diff
@@ -129,9 +132,30 @@ OGRFeature *OGRSOSILayer::GetNextFeature() {
     short nName, nNumLines;
     long  nNumCoo;
     unsigned short nInfo;
-     // TODO make a parameter that can be sent from command line
+
+    // Used to limit the number of limit the fields to be appended.
+    // Default is that duplicates will appended if no ogrSosiAppendFieldsMap parameter is sent.
     std::map<std::string,std::string> appendFieldsMap;
-    appendFieldsMap.insert(std::pair<std::string,std::string>("BEITEBRUKERID",","));
+
+	// Get ogrSosiAppendFieldsMap and update appendFieldsMap if present;
+    // The input must on this format
+    // -oo ogrSosiAppendFieldsMap="BEITEBRUKERID:,&FIELD2test: &FIELD3test:;"
+    // With the example abouve the field BEITEBRUKERID append character will be ':', for FIELD2test it will space and for FIELD3test it will be ;
+    const char * ogrSosiAppendFieldsMapInput = CSLFetchNameValue(poParent->GetOpenOptions(),"ogrSosiAppendFieldsMap");
```

Comment:
No need for ogr sosi, and regarding naming convention "APPEND_FIELDS_MAP" would match what is generally used